src/interfaces/chain.cpp

```diff
@@ -295,6 +296,14 @@ class ChainImpl : public Chain
         CInv inv(MSG_TX, txid);
         g_connman->ForEachNode([&inv](CNode* node) { node->PushInventory(inv); });
     }
+    bool broadcastTransaction(const CTransactionRef& tx, std::string& err_string, const CAmount& max_tx_fee, bool relay) override
+    {
+        const TransactionError err = BroadcastTransaction(tx, err_string, max_tx_fee, relay, /*wait_callback*/ false);
+        // Chain clients only care about failures to accept the tx to the mempool. Disregard non-mempool related failures.
+        // Note: this will need to be updated if BroadcastTransactions() is updated to return other non-mempool failures
+        // that Chain clients do not need to know about.
+        return TransactionError::OK == err || TransactionError::P2P_DISABLED == err;
```

Comment:
Might be better to do:

```
switch(err) {
    case TransactionError::OK:
    case TransactionError::P2P_DISABLED:
        return true;
    case TransactionError::MISSING_INPUTS:
    case TransactionError::ALREADY_IN_CHAIN:
    case TransactionError::MEMPOOL_REJECTED:
    case TransactionError::MEMPOOL_ERROR:
    case TransactionError::INVALID_PSBT:
    case TransactionError::PSBT_MISMATCH:
    case TransactionError::SIGHASH_MISMATCH:
    case TransactionError::MAX_FEE_EXCEEDED:
        return false;
}
```

That way you get a warning if there's other values added that you haven't handled.

Reply:
I don't think it really matters too much. If new TransactionError values are added that the wallet needs to handle, this interface can be updated.